core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -228,6 +261,46 @@ private[spark] class EventLoggingListener(
     fileSystem.rename(new Path(logPath + IN_PROGRESS), target)
   }
 
+  /**
+   * According to the updated event to modify the maintained event's metrics
+   * @param executorId  the executor whose metrics will be modified
+   */
+  private def updateModifiedMetrics(executorId: String): Unit = {
+    val toBeModifiedEvent = executorIdToModifiedMaxMetrics.get(executorId)
+    val latestEvent = executorIdToLatestMetrics.get(executorId)
+    toBeModifiedEvent match {
+      case None => if (latestEvent.isDefined) executorIdToModifiedMaxMetrics.update(
+        executorId, latestEvent.get)
```

Comment:
Bringing back a comment from an old diff -- I had mentioned that `latestEvent` should always be defined, and you responded:

> `latestEvent` should not always be defined when `toBeModifiedEvent.isEmpty` is true. The case is that at the beginning, the `latestEvent` is empty before the first metrics update event received.

I don't follow.  `updateModifiedMetrics` is only called from one place, `onExecutorMetricsUpdated`, L240.  And just before that, on L239, `executorIdToLatestMetrics(execId)` is set.  so I don't see how it could be undefined.  In fact, you could just change this method to take `latestEvent: SparkListenerExecutorMetricsUpdate` and then L240 becomes `updateModifiedMetrics(lightEvent)`, and here you get the executorId from that event.